zebra-rpc/src/methods.rs

```diff
@@ -526,6 +542,76 @@ where
         }
         .boxed()
     }
+
+    fn get_address_utxos(&self, addresses: Vec<String>) -> BoxFuture<Result<Vec<AddressUtxos>>> {
+        let mut state = self.state.clone();
+        let mut response_utxos = vec![];
+
+        async move {
+            // TODO: make this a function after #4062 is merged
+            let valid_addresses: Result<Vec<Address>> = addresses
+                .iter()
+                .map(|address| {
+                    address.parse().map_err(|_| {
+                        Error::invalid_params(format!("Provided address is not valid: {}", address))
+                    })
+                })
+                .collect();
+
+            // get utxos data for addresses
+            let request = zebra_state::ReadRequest::UtxosByAddresses(valid_addresses?);
+            let response = state
+                .ready()
+                .and_then(|service| service.call(request))
+                .await
+                .map_err(|error| Error {
+                    code: ErrorCode::ServerError(0),
+                    message: error.to_string(),
+                    data: None,
+                })?;
+            let utxos = match response {
+                zebra_state::ReadResponse::Utxos(utxos) => utxos,
+                _ => unreachable!("unmatched response to a UtxosByAddresses request"),
+            };
+
+            // for each utxo found, get transaction
+            for utxo in utxos {
+                // given hash, get transaction
+                let hash = utxo.output_location.hash;
+                let request = zebra_state::ReadRequest::Transaction(hash);
```

Comment:
Can be a problem to make this call for each utxo found, specially as we don't limit the amount of results. Pagination can be an option but `lightwalletd` does not currently support it.